### PR TITLE
fsspec: split the cloud-specific functionality from the base wrapper

### PR DESCRIPTION
**Pull request by isidentical**

### CODE REVIEW

File: dvc/fs/fsspec_wrapper.py [hunk 1]
```diff
@@ -164,3 +149,36 @@ def _download(
             ) as wrapped:
                 with open(to_file, "wb") as fdest:
                     shutil.copyfileobj(wrapped, fdest, length=fobj.blocksize)
+
+
+# pylint: disable=abstract-method
+class CloudFSSpecWrapper(FSSpecWrapper):
```
Comment (skshetry) [inline dvc/fs/fsspec_wrapper.py:155]: `Cloud` might be misleading (isn't `gdrive` in cloud?). How about `ObjectFSWrapper`? It is a bit of a mouthful though.

Comment (isidentical) [reply dvc/fs/fsspec_wrapper.py:155]: What about PrefixBasedFSWrapper?

Comment (skshetry) [reply dvc/fs/fsspec_wrapper.py:155]: @isidentical, I'd prefer `Object-*`, but no strong opinions.

Comment (isidentical) [reply dvc/fs/fsspec_wrapper.py:155]: Then Let's go with ObjectFSWrapper. Thanks for the suggestion!

**Review by karajan1001 — APPROVED** [review]

**Pull request merged**